DEPLOYMENT.md

```diff
@@ -7,32 +7,145 @@ On this page we list the various steps needed for deployments. The first section
+      }
+
+      location /media/ {
+      alias /data/freezeman/backend/media/;
+      }
+
+      location /api/ {
+      include uwsgi_params;
+      uwsgi_pass unix:/data/freezeman/backend/fms.sock;
+      uwsgi_read_timeout 300s;
+      }
+
+      location = /favicon.ico { access_log off; log_not_found off; }
+
+      error_page 404 /index.html;
+
+    systemctl restart nginx
+    ```
+  * Install certificate
+
+# Routine deployments
+  * Backup the database (`pg_dumpall > backup_release_vX_X_X.pgsql`)
+  * Kill the uwsgi processes (`ps -aux | grep uwsgi` to find, `kill -9 PROCESS_NUMBER` may need only kill the master)
+  * Move to the repository base directory (`cd ./freezeman`) and checkout the release tag from the repository (`git checkout vX.X.X`)
+  * Update the submodule pg_fzy (`git submodule update --init --recursive`) (if changed)
+  * Move to the frontend (`cd frontend`) and install any new dependency (`npm install`) and fix security issues (`npm audit fix`)
+  * Compile the frontend (`npm run build`)
+  * Move to the backend directory (`cd ../backend`) and activate the the virtual environment (`. env/bin/activate`)
+  * Install any new dependency (`pip3.8 install -r requirements.txt`) (version of pip may change)
+  * Install pg_fzy (`cd backend/dependencies/pg_fzy && make && sudo make install`) (`make restore_precompiled_binary` to get precompiled binaries) (if needed)
+  * Move back to the backend root (`cd ../..`) and migrate the database (`python3.8 manage.py migrate`)
+  * Create the first revisions for newly created models (`python3.8 manage.py createinitialrevisions`)
+  * Serve new and modified templates (`python3.8 manage.py collectstatic`)
+  * Activate the pg_fzy module (`psql -u postgres -d fms -c "create extension fzy;"`)
+  * Restart the uwsgi (`uwsgi uwsgi.ini &`)
+
+### Notes on submodule setup
+  * If the `git submodule update --init --recursive` command fails to reach the repo and set the submodule, check the inside the .git hidden directory of the repo base directory. Inside, open the config file and replace in the section [submodule "backend/dependencies/pg_fzy"], "url = git:" by "url = https:". Execute `git submodule update --init --recursive` then go into .git/modules/backend/dependencies/pg_fzy and open "config". Replace in the section [submodule "fzy_native"], "url = git:" by "url = https:". Execute `git submodule update --init --recursive` again.
+  * To update submodule to latest commit, `git submodule update --remote --merge`
+
+
+# Specific deployments
 
 * Version 3.1 : 
   * Upgrade python version to 3.8
   * Clone the new repository
   * Modify the nginx and uwsgi for the new locations.
 * Version 3.2 : 
-  * Add FREEZEMAN_SECRET_KEY to env variables through uwsgi.ini
\ No newline at end of file
+  * Add FMS_SECRETKEY, FMS_ENV, FMS_EMAIL_HOST, FMS_EMAIL_PORT, FMS_EMAIL_FROM, FMS_EMAIL_USER, FMS_EMAIL_USER, FMS_EMAIL_PASSWORD, FMS_EMAIL_TLS to env variables through uwsgi.ini
```

Comment:
Can you add details about the type of values these might be (i.e. FMS_ENV has to be 'PROD', 'QC' or 'DEV')